public/js/fish.js

```diff
@@ -29,6 +29,154 @@ if (lang && lang !== '' && lang.toLowerCase() in langs) {
+        }
+        // emitter.on() is cumulative! And showCatchIntentDialog() could be called multiple times...
+        // https://nodejs.org/api/events.html#emitteroneventname-listener
+        // NOTE: This is NOT the same as .once() !!!
+        $('#catch-intent-input').on('keydown', function (e) {
+            if (e.key === 'Enter' || e.keyCode === 13) {
+                recordMyCatchIntent();
+            }
+        });
+        $('#catch-intent-submit').on('click', recordMyCatchIntent);
+        myCatchIntentDialogConfigured = true;
+    }
+    $('#catch-intent-input').val("");
+    $('#catch-intent-submit').show();
+    $('#catch-intent-dialog-box').show();
+    $('#catch-intent-input').trigger('focus');
+}
+
+function hideCatchIntentDialog() {
+    $('#catch-intent-dialog-box').hide();
+}
+
+function checkCatchIntentDisplay() {
+    var season = myCatchIntentSubmitted ? st.catchIntentSeason : st.catchIntentDisplaySeason;
+    // console.log('checkCatchIntentDisplay(): ' + ' season=' + season + ' mySeason=' + myCatchIntentDisplaySeason);
+    if (season != myCatchIntentDisplaySeason) {
+        if (season == 0) {
+            hideCatchIntentColumn();
+        } else {
+            showCatchIntentColumn(season);
+        }
+        myCatchIntentDisplaySeason = season;
+    }
+}
+
+
+function startAskingIntendedCatch() {
+    showCatchIntentDialog();
+    myCatchIntent = '???';
+    myCatchIntentSubmitted = false;
+}
+
+function stopAskingIntendedCatch() {
+    hideCatchIntentDialog();
+}
+
+function recordMyCatchIntent() {
+    var input = $('#catch-intent-input').val().trim();
+    var num = parseInt(input);
+    if (isNaN(num) || num < 0) {
+        $('#catch-intent-input').val(myCatchIntent);
+        // Leave dialog up in case fisher corrects typo and resubmits
+    }
+    else {
+        myCatchIntent = num.toString();
+        stopAskingIntendedCatch();
+        submitMyCatchIntent();
+    }
+}
+
+function submitMyCatchIntent() {
+    socket.emit('recordIntendedCatch', myCatchIntent);
+    myCatchIntentSubmitted = true;
+}
+
+
+////////////////////////////////////////
+//////////// END Catch Intentions feature   (except for a few touch points below) 
+////////////////////////////////////////
+
+////////////////////////////////////////
+//////////// START Profit Colum Display Feature     //Profit.Vis
+////////////////////////////////////////
+
+//controls visibility of both seasonal and overall profit columns in one function for show and one for hide
+//for tutorial text, table column heading, and table body
+
+// Not needed now. Maybe later?
+// function showProfitColumns(season) {
+//     $('#profit-season-header').show();
+//     $('#profit-season-th').show();
+//     $('#profit-total-header').show();
+//     $('#profit-total-th').show();
+//     for (var i in st.fishers) {
+//         $('#f' + i + '-profit-season').show();
+//         $('#f' + i + '-profit-total').show();
+//     }
+// }
```

Comment:
I'd err on the side of not including this commented function if it's not needed now 🙂